### PR TITLE
teleinfo: update documentation since sensor/text_sensor move.

### PR DESCRIPTION
**Pull request by 0hax**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1403

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/teleinfo.rst [hunk 1]
```diff
@@ -97,6 +104,15 @@ Configuration variables:
 - **uart_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the :ref:`UART Component <uart>` if you want
   to use multiple UART buses.
 
+In a teleinfo sensor, in addition to normal sensor configuration: :ref:`Sensor <config-sensor>`
+
+- **tag_name** (**Required**): Specify the tag you want to retrieve from the Teleinformation and select with what name to transmit it.
```
Comment (glmnet) [inline components/sensor/teleinfo.rst]: The name to transmit it is the sensor name itself? not the name of the tag_name, right?
```suggestion
- **tag_name** (**Required**, string): Specify the tag you want to retrieve from the Teleinformation.
- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
- All other options from :ref:`Sensor <config-sensor>`.
```

File: components/sensor/teleinfo.rst [hunk 2]
```diff
@@ -97,6 +104,15 @@ Configuration variables:
...
+
+In a text sensor, in addition to normal text sensor configuration: :ref:`Text Sensor <config-text_sensor>`
+
+- **tag_name** (**Required**): Specify the tag you want to retrieve from the Teleinformation and select with what name to transmit it.
```
Comment (glmnet) [inline components/sensor/teleinfo.rst:113]: The name to transmit it is the sensor name itself? not the name of the tag_name, right?
```suggestion
- **tag_name** (**Required**): Specify the tag you want to retrieve from the Teleinformation.
```

Comment (glmnet) [inline components/sensor/teleinfo.rst]: The name to transmit it is the sensor name itself? not the name of the tag_name, right?
```suggestion
- **tag_name** (**Required**, string): Specify the tag you want to retrieve from the Teleinformation.
- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
- All other options from :ref:`Text Sensor <config-text-sensor>`.
```

File: components/sensor/teleinfo.rst [hunk 3]
```diff
@@ -97,6 +104,15 @@ Configuration variables:
 - **uart_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the :ref:`UART Component <uart>` if you want
   to use multiple UART buses.
 
+In a teleinfo sensor, in addition to normal sensor configuration: :ref:`Sensor <config-sensor>`
```
Comment (glmnet) [inline components/sensor/teleinfo.rst]:
```suggestion
Sensor
******
```

File: components/sensor/teleinfo.rst [hunk 4]
```diff
@@ -97,6 +104,15 @@ Configuration variables:
...
+
+- **tag_name** (**Required**): Specify the tag you want to retrieve from the Teleinformation and select with what name to transmit it.
+
+In a text sensor, in addition to normal text sensor configuration: :ref:`Text Sensor <config-text_sensor>`
```
Comment (glmnet) [inline components/sensor/teleinfo.rst]:
```suggestion
Text Sensor
***********
```

**Review by glmnet — COMMENTED** [review]
Add some docs standard

**Review by jesserockz — APPROVED** [review]

**Pull request merged**